File: Diagrammatize.Design.Pattern/code/python/Builder/Sample/TextBuilder.py
import io

class TextBuilder:
    def __init__(self):
        self.buffer = io.StringIO()       # 文档内容保存在该字段中 

    def makeTitle(self, title):                         # 纯文本的标题
        self.buffer.write("==============================\n")       # 装饰线
        self.buffer.write("『" + title + "』\n")                    # 为标题加上『』
        self.buffer.write("\n")                                     # 换行


    def makeString(self, str_):                         # 纯文本的字符串
        self.buffer.write("■" + str_ + "\n")                        # 为字符串添加■
        self.buffer.write("\n")                                     # 换行

    def makeItems(self, items):                         # 纯文本的条目
        for i in range(len(items)):
            self.buffer.write("　・" + items[i] + "\n")             # 为条目添加・
        self.buffer.write("\n")                                     # 换行

    def close(self):                                    # 完成文档
        self.buffer.write("==============================\n")       # 装饰线

    def getResult(self):                                # 完成后的文档
        return self.buffer.getvalue()

File: Diagrammatize.Design.Patterns/code/cxx/Builder/A4/TextBuilder.cpp
#include "TextBuilder.hpp"

TextBuilder::~TextBuilder() {
}

void TextBuilder::makeTitle(const std::string& title) {     // 纯文本的标题
    buffer += "==============================\n";               // 装饰线
    buffer += "『" + title + "』\n";                            // 为标题加上『』
    buffer += "\n";                                             // 换行
}

void TextBuilder::makeString(const std::string& str) {      // 纯文本的字符串
    buffer += "■" + str + "\n";                                // 为字符串添加■
    buffer += "\n";                                             // 换行
}

void TextBuilder::makeItems(const std::vector<std::string>& items) {    // 纯文本的条目
    for (int i = 0; i < items.size(); i++) {
        buffer += "　・" + items[i] + "\n";                     // 为条目添加・
    }
    buffer += "\n";                                             // 换行
}

void TextBuilder::close() {                             // 完成文档
    buffer += "==============================\n";               // 装饰线
}

std::string TextBuilder::getResult() {                  // 完成的文档
    return buffer;
}
